go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -291,12 +291,20 @@ func allocatePodAnnotationWithRollback(
 
 	var ipamClaim *ipamclaimsapi.IPAMClaim
 	hasIPAMClaim := hasIPAM && claimsReconciler != nil && network != nil && network.IPAMClaimReference != ""
-	if hasIPAMClaim {
+	if netInfo.AllowsPersistentIPs() && hasIPAMClaim {
 		ipamClaim, err = claimsReconciler.FindIPAMClaim(network.IPAMClaimReference, network.Namespace)
 		if err != nil {
 			err = fmt.Errorf("error retrieving IPAMClaim for pod %s/%s: %w", pod.GetNamespace(), pod.GetName(), err)
 		}
 		hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
+	} else if hasIPAMClaim {
+		klog.Errorf(
+			"Pod %s/%s requesting an IPAMClaim reference on network %q which does not allow it",
+			pod.GetNamespace(),
+			pod.GetName(),
+			netInfo.GetNetworkName(),
+		)
+		hasIPAMClaim = false
```

Comment:
Wouldn't it just be cleaner to do something like
```
hasPersistentIPs := netInfo.AllowsPersistentIPs() && hasIPAM && claimsReconciler != nil
hasIPAMClaim := network != nil && network.IPAMClaimReference != ""
if hasIPAMClaim && !hasPersistentIPs {
    klog.Errorf(
        "Pod %s/%s referencing an IPAMClaim on network %q which does not honor it",
	pod.GetNamespace(),
	pod.GetName(),
	netInfo.GetNetworkName(),
    )
    hasIPAMClaim = false
}
if hasIPAMClaim {
   ipamClaim, err = claimsReconciler.FindIPAMClaim(network.IPAMClaimReference, network.Namespace)
   if err != nil  {
      err = fmt.Errorf("error retrieving IPAMClaim for pod %s/%s: %w", pod.GetNamespace(), pod.GetName(), err)
      return
   }
   hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
}
```

Reply:
notice the missing return on error in your code that I included in my code excerpt

Reply:
Right! Done.